src/netaddress.cpp

```diff
@@ -695,7 +696,7 @@ uint32_t CNetAddr::GetLinkedIPv4() const
         // Teredo tunneled IPv4: the IPv4 address is in the last 4 bytes of the address, but bitflipped
         return ~ReadBE32(Span{m_addr}.last(ADDR_IPV4_SIZE).data());
     }
-    assert(false);
+    UNREACHABLE();
```

Comment:
This is now also changing code that does not use switch statements, and I assume doesn't warn under mingw-w64, becoming inconsistent with the rest of the codebase. If we are going to change some of these, we should be changing all, otherwise this is just random refactoring.

Reply:
> I assume doesn't warn under mingw-w64

It does when compiling with `-O0`.

Reply:
There'd be test/lint/lint-assertions.py if this is a goal

Reply:
@fanquake 
> we should be changing all, otherwise this is just random refactoring.

In that case, what macro name could be appropriate?